src/rust/grapl-aws/src/s3/client.rs

```diff
@@ -0,0 +1,82 @@
+use rusoto_s3::S3;
+
+use crate::s3::bucket::{Bucket, BucketBuilder};
+
+#[derive(Clone)]
+pub struct S3Client<S>
+where
+    S: S3Common,
+{
+    #[cfg(not(test))]
+    pub(crate) s3_client: S,
+    #[cfg(test)]
+    pub s3_client: S,
```

Comment:
Can we call this member something like `inner`? Otherwise we end up with code like `s3_client.s3_client.do_something()`, which looks kinda weird and makes you do a double take.